File: 2/find_by_name_and_size.c
#include <dir_each.h>
#include <stdio.h>
#include <stdlib.h>
#include <sys/stat.h>
#include <string.h>

void find(char *dir_name, char *find_str, int min_byte){
    void compare_name_and_size(struct dirent * dp){
		if(strncmp(dp->d_name, find_str, strlen(find_str))==0){
			struct stat	statbuf;
			char buf[1024]={0};
			strcat(buf,dir_name); 
			strcat(buf,dp->d_name);
			if (stat(buf, &statbuf) !=-1){
				if(statbuf.st_size>=min_byte) printf("%s, %d\n",dp->d_name, (int)statbuf.st_size);
			}
		}
	}
	each(compare_name_and_size, dir_name);
}

int main(int argc, char *argv[]){
	if(argc<=3) find("./","a",10);
	else find(argv[1],argv[2],atoi(argv[3]));
	return 1;
}

File: 1/find_by_name_and_size.rb
def find(dir,find_str,min_byte)
	Dir.open(dir).each do |x|
		full_path = dir+x
		puts "#{x}, #{File.size(full_path)}" if x.start_with?(find_str) && File.size(full_path)>min_byte
	end
end

if ARGV.length==3
	find(ARGV[0],ARGV[1],ARGV[2].to_i)
else
	find("./", "a", 10)
end

